tests/ut/ir/transforms/test_comments_survive_mutator.py

```
@@ -0,0 +1,70 @@
+        Documented v1 limitation: IRMutator *subclasses* that construct stmts
+        directly (bypassing MakeLikeStmt) may drop comments. The ``convert_to_ssa``
+        pass goes through the base path for ``ForStmt`` but not for ``AssignStmt``;
+        this test only asserts the preservation that the base path guarantees.
+        """
+
+        @pl.program
+        class P:
+            @pl.function
+            def main(self, x: pl.Scalar[pl.FP32]) -> pl.Scalar[pl.FP32]:
+                # loop annotation
+                for i in pl.range(4):  # trailing
```

Ruff (0.15.9)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)